plugins/Goals/javascripts/goalsForm.js

```diff
@@ -29,7 +29,8 @@ function hideForms() {
 
 function showCancel() {
     $(".entityCancel").show();
-    $('.entityCancelLink').click(function () {
+    $('.entityCancelLink').off('click').click(function () {
```

Comment:
We always attached a new event listener when opening a goal but instead need to remove previously added once first